### PR TITLE
added GET categories Q3

### PR DESCRIPTION
**Pull request by Greeny1248**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,61 +1,3 @@
...
-
-## Husky
-
-To ensure we are not commiting broken code this project makes use of git hooks. Git hooks are scripts triggered during certain events in the git lifecycle. Husky is a popular package which allows us to set up and maintain these scripts. This project makes use a _pre-commit hook_. When we attempt to commit our work, the script defined in the `pre-commit` file will run. If any of our tests fail than the commit will be aborted.
-
-The [Husky documentation](https://typicode.github.io/husky/#/) explains how to configure Husky for your own project as well as creating your own custom hooks.\_
-
 ## Adding ENV files to access databases
 
 You will need to create two .env files for your project: .env.test and .env.development. Into each, add PGDATABASE=<database_name_here>, with the correct database name for that environment (see /db/setup.sql for the database names).
```
Comment (MKD01) [inline README.md:3]: Changes to the README.md should have been in a seperate pr for task 2

File: __tests__/app.test.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      test("returns an array of objects with a slug and desc", () => {
+        return request(app)
+          .get("/api/categories")
+          .expect(200)
+          .then((res) => {
+            expect(typeof res.body.categories[0]).toBe("object");
+            expect(res.body.categories[0].hasOwnProperty("description")).toBe(
+              true
+            );
+            expect(Array.isArray(res.body.categories)).toBe(true);
```
Comment (MKD01) [inline __tests__/app.test.js]: These test assertions can be cleaned up a little, the typeof and Array.isArray is not directly needed as this is indirectly tested for in the expect for hasOwnProperty. 

However this testing is just asserting that the first element of categories has a property called description, but instead should be that every element in categories should have this property as well as a slug property.

There should also be an assertion that also checks that categories has atleast 1 element within in.

File: app.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+const express = require("express");
+const app = express();
+app.use(express.json());
```
Comment (MKD01) [inline app.js]: This is not needed for this end point

File: controller/controller.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+const { devData } = require("../db/data/development-data/index");
+const { readCategories } = require("../model/model.js");
+
+const viewAllCategories = (req, res) => {
+  readCategories().then((categories) => {
+    res.status(200).send({ categories });
+  });
```
Comment (MKD01) [inline controller/controller.js]: There should be a .catch() here which deals with potential errors

File: model/model.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+const db = require("../db/connection");
+
+readCategories = () => {
+  let queryString = `SELECT * FROM categories`;
+  return db
+    .query(queryString)
+    .then((results) => {
+      console.log(results.rows);
```
Comment (MKD01) [inline model/model.js:9]: console.log is not needed

File: model/model.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+      return results.rows;
+    })
+    .catch((err) => {
+      console.log(err);
+    });
```
Comment (MKD01) [inline model/model.js:14]: The .catch is not needed here but can be added in the controller.

File: app.js [hunk 7]
```diff
@@ -0,0 +1,8 @@
...
+const { viewAllCategories } = require("./controller/controller");
+
+app.get("/api/categories", viewAllCategories);
+
```
Comment (MKD01) [inline app.js:20]: There can also be error handling added her for example if a user tries make a request to a non existant end point, which should also be tested for.

**Review by MKD01 — APPROVED** [review]
Im happy for this to be merged once those changes have been made.

File: app.js (hunk 3 above)
Comment (Greeny1248) [reply app.js]: Do you just mean the express.json part? as i need to use "app"

Comment (Greeny1248) [reply app.js]: fixed

File: app.js (hunk 7 above)
Comment (Greeny1248) [reply app.js:20]: added 404 and 500 error messages (struggling to test for 500 error though)

**Pull request merged**